siliconcompiler/flows/fpgaflow.py

```diff
@@ -58,25 +58,30 @@ def setup(chip, flowname='fpgaflow'):
 
     '''
 
-    # Check that partname has been set
-
-    if chip.get('fpga', 'partname'):
+    # Check that fpga arch has been set for vpr flow or partname has been set for others
+    flow = ''
+    if chip.get('fpga', 'arch'):
+        flow = 'vpr'
+    elif chip.get('fpga', 'partname'):
         partname = chip.get('fpga', 'partname')
     else:
         chip.logger.error("FPGA partname not specified")
         raise siliconcompiler.SiliconCompilerError("FPGA partname not specified")
 
     # Set FPGA mode if not set
     chip.set('option', 'mode', 'fpga')
-
-    # Partname lookup
-    (vendor, flow) = flow_lookup(partname)
-    chip.set('fpga', 'vendor', vendor)
+    
+    # Partname lookup for flows other than vpr   
+    if flow != 'vpr':
+        (vendor, flow) = flow_lookup(partname)
+        chip.set('fpga', 'vendor', vendor)
 
     #Setting up pipeline
     #TODO: Going forward we want to standardize steps
     if  flow in ('vivado', 'quartus'):
         flowpipe = ['syn', 'place', 'route', 'bitstream']
+    elif flow =='vpr':
+        flowpipe = ['syn', 'pack-place-route']
```

Comment:
No need to change this now, but going forward, I think this should be split up into separate steps for [pack, place, route, analyze, bistream]